apps/voting/app/src/screens/Votes.js

```diff
@@ -1,17 +1,296 @@
+            (outcomeFilter === 4 &&
+              voteStatus === VOTE_STATUS_ACCEPTED &&
+              isVoteAction(vote) &&
+              !executed))) &&
+        // app
+        // date range
+        (!(start || end) ||
+          isWithinInterval(new Date(startTimestamp), {
+            start: startOfDay(start),
+            end: endOfDay(end),
+          }) ||
+          isWithinInterval(endDate, {
+            start: startOfDay(start),
+            end: endOfDay(end),
+          }))
+      )
+    })
+    setFilteredVotes(filtered)
+  }, [
+    statusFilter,
+    outcomeFilter,
+    trendFilter,
+    appFilter,
+    dateRangeFilter,
+    pctBase,
+    setFilteredVotes,
+    votes,
+  ])
+
+  return {
+    filteredVotes,
+    voteStatusFilter: statusFilter,
+    handleVoteStatusFilterChange: useCallback(
+      index => {
+        setStatusFilter(!index ? -1 : index)
+        setTrendFilter(-1)
+      },
+      [setStatusFilter, setTrendFilter]
+    ),
+    voteOutcomeFilter: outcomeFilter,
+    handleVoteOutcomeFilterChange: useCallback(
+      index => setOutcomeFilter(!index ? -1 : index),
+      [setOutcomeFilter]
+    ),
+    voteTrendFilter: trendFilter,
+    handleVoteTrendFilterChange: useCallback(
+      index => setTrendFilter(!index ? -1 : index),
+      [setTrendFilter]
+    ),
+    voteAppFilter: appFilter,
+    handleVoteAppFilterChange: useCallback(
+      index => setAppFilter(!index ? -1 : index),
+      [setAppFilter]
+    ),
+    voteDateRangeFilter: dateRangeFilter,
+    handleVoteDateRangeFilterChange: setDateRangeFilter,
+    handleClearFilters,
+  }
+}
 
+const useVotes = votes => {
+  const sortedVotes = votes.sort(sortVotes)
   const openVotes = sortedVotes.filter(vote => vote.data.open)
   const closedVotes = sortedVotes.filter(vote => !openVotes.includes(vote))
-  const votingGroups = [['Open votes', openVotes], ['Past votes', closedVotes]]
+  return { openVotes, closedVotes }
+}
+
+const LayoutVotes = React.memo(function LayoutVotes({
+  votes,
+  selectedVote,
+  selectVote,
+  onVote,
+  onExecute,
+}) {
+  const theme = useTheme()
+  const { layoutName } = useLayout()
+  const {
+    filteredVotes,
+    voteStatusFilter,
+    handleVoteStatusFilterChange,
+    voteOutcomeFilter,
+    handleVoteOutcomeFilterChange,
+    voteTrendFilter,
+    handleVoteTrendFilterChange,
+    voteAppFilter,
+    handleVoteAppFilterChange,
+    voteDateRangeFilter,
+    handleVoteDateRangeFilterChange,
+    handleClearFilters,
+  } = useFilterVotes(votes, votes.map(({ voteId }) => voteId))
+  const { openVotes, closedVotes } = useVotes(filteredVotes)
+  const handleBackClick = () => selectVote(-1)
+
+  return (
+    <React.Fragment>
+      {selectedVote !== null && (
+        <React.Fragment>
+          <Bar>
+            <BackButton onClick={handleBackClick} />
+          </Bar>
+          <Vote vote={selectedVote} onVote={onVote} onExecute={onExecute} />
+        </React.Fragment>
```

Comment:
I avoided refactoring the code to focus on the new styles so that this does not push back the release but for sure, we could refactor the code, maybe after the release?

Reply:
Done in 8928076 and fe92175.